src/core_scan/Core_scan.ml

```diff
@@ -702,57 +710,60 @@ let iter_targets_and_get_matches_and_exn_to_errors (config : Core_scan_config.t)
 (* File targeting and rule filtering *)
 (*****************************************************************************)
 
-let xtarget_of_file ~parsing_cache_dir (xlang : Xlang.t) (file : Fpath.t) :
-    Xtarget.t =
-  let lazy_ast_and_errors =
-    lazy
-      (let lang =
-         (* ew. We fail tests if this gets pulled out of the lazy block. *)
-         match xlang with
-         | L (lang, []) -> lang
-         | L (_lang, _ :: _) ->
-             failwith
-               "xlang from the language field in -target should be unique \
-                (this shouldn't happen FIXME)"
-         | _ ->
-             (* alt: could return an empty program, but better to be defensive*)
-             failwith
-               "requesting generic AST for an unspecified target language"
-       in
-       Parse_with_caching.parse_and_resolve_name ~parsing_cache_dir
-         AST_generic.version lang file)
-  in
-  {
-    Xtarget.file;
-    xlang;
-    lazy_content = lazy (UFile.read_file file);
-    lazy_ast_and_errors;
-  }
-
-let lockfile_target_of_input_to_core
-    ({ path; lockfile_kind; manifest_target } : In.lockfile_target) =
+let lockfile_target_of_location
+    ({ source; file; kind = lockfile_kind; manifest } :
+      Target_location.lockfile) =
   let manifest_target =
-    manifest_target
-    |> Option.map @@ fun In.{ path; manifest_kind } ->
-       let path = Fpath.v path in
+    manifest
+    |> Option.map
+       @@ fun ({ source; file; kind = manifest_kind } :
+                Target_location.manifest) : Lockfile_target.manifest_target ->
        {
-         Lockfile_target.manifest = path;
-         lazy_manifest_content = lazy (UFile.read_file path);
-         lazy_manifest_ast_and_errors =
-           lazy (Parse_lockfile.parse_manifest manifest_kind path);
-         manifest_kind = Manifest_kind.of_lockfile_kind lockfile_kind;
+         source;
+         file;
+         lazy_content = lazy (UFile.read_file file);
+         lazy_ast_and_errors =
+           lazy (Parse_lockfile.parse_manifest manifest_kind file);
+         kind = Manifest_kind.of_lockfile_kind lockfile_kind;
        }
   in
-  let path = Fpath.v path in
   {
-    Lockfile_target.lockfile = path;
-    lockfile_kind;
-    lazy_lockfile_content = lazy (UFile.read_file path);
-    lazy_lockfile_ast_and_errors =
-      lazy (Parse_lockfile.parse_lockfile lockfile_kind manifest_target path);
-    manifest_target;
+    Lockfile_target.source;
+    file;
+    kind = lockfile_kind;
+    lazy_content = lazy (UFile.read_file file);
+    lazy_ast_and_errors =
+      lazy (Parse_lockfile.parse_lockfile lockfile_kind manifest_target file);
+    manifest = manifest_target;
   }
 
+let manifest_target_location_of_input_to_code
```

Comment:
Do these helpers belong here? I think I'd rather have them in Target_location or a separate module.

Reply:
I believe there is a dependency issue there. This is in `Core_scan` (in `semgrep.core_scan`), and `Target_location` is in `semgrep.core`, but `Input_to_core` is available via `semgrep.targeting`. I would hesitate to move that without understanding why that is the case.